rotkehlchen/externalapis/covalent.py

```diff
@@ -0,0 +1,260 @@
+        )
+        # TODO input and nonce is decoded in Covalent api, encoded in future
+        return CovalentTransaction(
+            timestamp=timestamp,
+            block_number=data['block_height'],
+            tx_hash=data['tx_hash'],
+            from_address=data['from_address'],
+            to_address=data['to_address'],
+            value=read_integer(data, 'value', DEFAULT_API),
+            gas=read_integer(data, 'gas_offered', DEFAULT_API),
+            gas_price=read_integer(data, 'gas_price', DEFAULT_API),
+            gas_used=read_integer(data, 'gas_spent', DEFAULT_API),
+            input_data='0x',
+            nonce=0,
+        )
+    except KeyError as e:
+        raise DeserializationError(
+            f'Covalent avalanche transaction missing expected key {str(e)}',
+        ) from e
+
+
+class Covalent():
+    """ChainID
+    - Ethereum 	            1
+    - Polygon/Matic 	    137
+    - Avalanche C-Chain     43114
+    - Binance Smart Chain   56
+    - Fantom Opera 	        250
+    - RSK 	                30
+    - Arbitrum 	            42161
+    - Palm 	                11297108109
+    """
+    def __init__(self,
+                 msg_aggregator: MessagesAggregator,
+                 chain_id: int) -> None:
+        self.session = requests.session()
+        self.session.headers.update({'User-Agent': 'rotkehlchen'})
+        self.msg_aggregator = msg_aggregator
+        self.chain_id = chain_id
+
+    def _query(
+            self,
+            module: str,
+            action: str,
+            address: str = None,
+            options: Optional[Dict[str, Any]] = None,
+            timeout: Optional[Tuple[int, int]] = None,
+    ) -> Optional[Dict[str, Any]]:
+        """Queries Covalent
+
+        May raise:
+        - RemoteError if there are any problems with reaching Covalent or if
+        an unexpected response is returned
+        """
+        query_str = f'https://api.covalenthq.com/v1/{self.chain_id}/{action}'
+        if address:
+            query_str += f'/{address}'
+        query_str += f'/{module}/'
+
+        query_str += f'?key={KEY}'
+
+        if options:
+            for name, value in options.items():
+                query_str += f'&{name}={value}'
+
+        retry = 0
+        while retry <= CONST_RETRY:
+            logger.debug(f'Querying covalent: {query_str}')  # noqa: E501 lgtm [py/clear-text-logging-sensitive-data]
+            try:
+                response = self.session.get(query_str, timeout=timeout if timeout else DEFAULT_TIMEOUT_TUPLE)  # noqa: E501
+            except requests.exceptions.RequestException as e:
+                # In timeout retry
+                if isinstance(e, requests.exceptions.Timeout):
+                    if retry == CONST_RETRY:
+                        return None
+                    retry += 1
+                    continue
+                raise RemoteError(f'Covalent API request failed due to {str(e)}') from e
+
+            result = response.json()
```

Comment:
You will need to also handle json decode error here, in case covalent returns something other than json due to some reason and raise `RemoteError` when that happens.

We do this everywhere. Example here: https://github.com/rotki/rotki/blob/3d5c17cbd598bbfa10a7b011c69f419a1da2265e/rotkehlchen/externalapis/etherscan.py#L204-L210